app/src/main/java/co/nimblehq/di/module/DataModule.kt

```diff
@@ -64,36 +55,4 @@ class DataModule {
             .addCallAdapterFactory(CoroutineCallAdapterFactory())
             .build()
     }
-
-    private fun createOkHttpClientBuilder(): OkHttpClient.Builder = OkHttpClient.Builder()
-        .readTimeout(30000, TimeUnit.MILLISECONDS)
-        .connectTimeout(30000, TimeUnit.MILLISECONDS)
-        .pingInterval(30, TimeUnit.SECONDS)
-        .addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
-
-}
-
-class AuthorizationInterceptor(
-    private val account: SurveysAccount,
-    private val tokenRepository: TokenRepository
-) : Interceptor {
-
-    override fun intercept(chain: Interceptor.Chain): Response {
-        val request = setAuthorizationHeader(chain.request())
-        var response = chain.proceed(request)
-        if (response.code() == 401) { //if unauthorized
```

Comment:
hi @ViktorArtemiev,
I'm not sure if it was due to my suggestion before that misleads you a little bit with the replacement of Authenticator to Interceptor, so I'd like to clarify this specific issue again.

As for this removal, it looks fine, the verification part (check 401 status) is delegated to the `Authenticator`.

But I'm observing that you're also removing entirely the rest of this Interceptor, which is causing a new severe issue with the performance:

- The app always starts *any network request* with a `401` result because there is no default place where you apply your cached Token to, which is meant to be done with this function `setAuthorizationHeader()`.
- The authenticator works correctly, it refreshes a new token, cached and using it for the next sequential request..